src/elements/swapMarket/SwapMarket.tsx

```diff
@@ -224,35 +224,35 @@ export const SwapMarket = ({
             errorMsg: `Trading ${fromAmount} ${fromToken.symbol} for ${toAmount} ${toToken.symbol} had failed. Please try again or contact support`,
           },
           txHash,
-        })
-      );
-    } catch (e) {
-      console.error('Swap failed with error: ', e);
-      if (e.code === ErrorCode.DeniedTx)
-        dispatch(
-          addNotification({
-            type: NotificationType.error,
-            title: 'Transaction Rejected',
-            msg: 'You rejected the trade. If this was by mistake, please try again.',
-          })
-        );
-      else {
-        const conversion = getConversionLS();
+        }),
```

Comment:
same here - components keep growing and contain lots of logic. We should try to keep only structural data in components (template), all logic that's bound to the component life cycle should go in custom hooks, and all the logic that can be extracted to live outside of the scope of a react component - should be extracted (to helper functions, services, etc.)